src/components/App.jsx

```diff
@@ -0,0 +1,209 @@
+import "../blocks/App.css";
+import { useState, useEffect } from "react";
+import { Route, Routes, Navigate, useNavigate } from "react-router-dom";
+import { SearchContext } from "../contexts/SearchContext";
+import { UserContext } from "../contexts/UserContext";
+import { AppContext } from "../contexts/AppContext";
+import Api from "../utils/api";
+import NewsApi from "../utils/newsApi";
+import { authorize } from "../utils/auth";
+import Nav from "./Nav";
+import Main from "./Main";
+import Footer from "./Footer";
+import ProtectedRoute from "./ProtectedRoute";
+import SavedNews from "./SavedNews";
+import SigninModal from "./SigninModal";
+import SignupModal from "./SignupModal";
+
+export default function App() {
+  const [searchState, setSearchState] = useState({
+    results: [],
+    keyword: "",
+    articlesAvail: 0,
+    articlesShown: 3,
+    loading: false,
+    nothingFound: false,
+  });
+  const [userState, setUserState] = useState({
+    loggedIn: false,
+    email: "",
+    username: "",
+    savedNews: [],
+  });
+  const [activeModal, setActiveModal] = useState("");
+  const [protectedDestination, setProtectedDestination] = useState("");
+  const navigate = useNavigate();
+
+  const api = new Api();
+
+  const newsApi = new NewsApi({
+    baseUrl: "https://nomoreparties.co/news/v2/everything",
+    apiKey: "a16de474931b4e5a83f83ad53ba3df69",
+  });
+
+  useEffect(() => {
+    if (protectedDestination !== "") setActiveModal("signin");
+  }, [protectedDestination]);
+
+  useEffect(() => {
+    document.addEventListener("keydown", (e) => {
```

Comment:
You should add an `Escape` listener like this to be able to remove it **correctly** when modals are closed:

```jsx
   useEffect(() => {

    if (!activeModal) return; // stop the effect not to add the listener if there is no active modal

    const handleEscClose = (e) => {  // define the function inside useEffect not to lose the reference on rerendering
      if (e.key === "Escape") {
        handleCloseModal();
      }
    };

    document.addEventListener("keydown", handleEscClose);

    return () => {  // don't forget to add a clean up function for removing the listener
      document.removeEventListener("keydown", handleEscClose);
    };
  }, [activeModal]);  // watch activeModal here
```

Here you define ` handleEscClose` inside the `useEffect` so that it could maintain a **fixed** reference if the component rerenders.
And of course we need to return a `clean up` function to **`remove`** the listener if the component unmounts.
And don’t forget about the **`dependencies array`** in order to watch `activeModal`